run.py

```diff
@@ -906,119 +905,93 @@ def hvreads(args):
-                  read, = reads
-                  if len(read[0]) > MIN_READ_LENGTH_FOR_ALIGNMENT:
-                     combined.append((seq_id, read))
-               else:
-                  read1, read2 = reads
-
-                  read1_long_enough = len(
-                     read1[0]) > MIN_READ_LENGTH_FOR_ALIGNMENT
-                  read2_long_enough = len(
-                     read2[0]) > MIN_READ_LENGTH_FOR_ALIGNMENT
-
-                  if read1_long_enough and read2_long_enough:
-                     pair1.append((seq_id, read1))
-                     pair2.append((seq_id, read2))
-                  elif read1_long_enough:
-                     combined.append((seq_id, read1))
-                  elif read2_long_enough:
-                     combined.append((seq_id, read2))
-
-         for label, reads in [
-               ("pair1", pair1),
-               ("pair2", pair2),
-               ("combined", combined),
-         ]:
-            if reads:
-               with open("%s.fastq" % label, "w") as outf:
-                  for seq_id, read in reads:
-                     seq, quality = read
-                     outf.write("@%s\n%s\n+\n%s\n" % (seq_id, seq, quality))
-
-         assert len(pair1) == len(pair2)
-
-         if not pair1 and not combined:
-            continue
+            output = potential_input.replace(".gz", ".alignments.tsv")
+            inputs = [potential_input]
+            if ".pair1." in output:
+               output = output.replace(".pair1.", ".")
+               inputs.append(potential_input.replace(".pair1.", ".pair2."))
+            elif ".pair2" in output:
+               # We handle pair1 and pair2 together.
+               continue
 
-         sam_out = "out.sam"
-         cmd = [
-            "/home/ec2-user/bowtie2-2.5.2-linux-x86_64/bowtie2",
-            "--local",
-            "-x", os.path.join(THISDIR, "bowtie", "human-viruses"),
-            "--very-sensitive-local",
-            "--score-min", "G,1,0",
-            "--mp", "2,0",
-            "--no-unal",
-            "--threads", "24",
-            # Including an @SQ line makes the .sam files really big, and we
-            # don't actually need to know the lengths of all the genomes we
-            # considered aligning against. When we have a small number of reads
-            # per sample this can bloat the files 100x.  We'd like to do
-            # "--no-sq" bur pysam chokes on SAM files that are missing the @SQ
-            # header lines.
-            "-S", sam_out,
-         ]
-
-         assert bool(pair1) == bool(pair2)
-         if pair1:
-            cmd.extend(["-1", "pair1.fastq", "-2", "pair2.fastq"])
-         if combined:
-            cmd.extend(["-U", "combined.fastq"])
+            outputs.append(output)
 
-         subprocess.check_call(cmd)
+            # TODO(jefftk): do we have a problem when read1 or read2 is too
+            # short?  I remember Bowtie choking on these before.
```

Comment:
FYI I'm currently running this on a bunch of sample files to learn if this is a problem